chia/plotting/manager.py

```diff
@@ -392,6 +392,9 @@ def process_file(file_path: Path) -> Optional[PlotInfo]:
                         if not self.open_no_key_filenames:
                             return None
 
+                    if file_path in self.no_key_filenames:
+                        self.no_key_filenames.remove(file_path)
```

Comment:
it's not so easy to follow what's going on here. there's a lot of state being mutated back and forth.

It seem like we *add* `file_path` to the `self.no_key_filenames` list just a few lines above, and then we remove it if we get here.. it seems like it would be much simpler to not add it in the first place. on line 391 and 377, we add this file path, and then sometimes return. If we don't return, we get here and we remove it.

Wouldn't it be simpler to just add the path in the case where we return instead? or is there some subtle interaction I'm missing?

i.e. on line 391 and 377, instead say something like:

```
if not self.open_no_key_filenames:
   self.no_key_filenames.add(file_path)
   return None
```

Reply:
I added a comment to make it clear, see 71d59f9. If i get your suggestion right, this would work if we would clear `no_key_filenames` at the start of each refresh interval.

Reply:
I see. thanks!